sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py

```diff
@@ -0,0 +1,299 @@
+
+# The test class name needs to start with "Test" to get collected by pytest
+class TestagentClient(AzureRecordedTestCase):
+
+    # helper function: create client using environment variables
+    def create_client(self, **kwargs):
+        # fetch environment variables
+        connection_string = kwargs.pop("azure_ai_projects_connection_string_agents_tests")
+        credential = self.get_credential(AIProjectClient, is_async=False)
+
+        # create and return client
+        client = AIProjectClient.from_connection_string(
+            credential=credential,
+            conn_str=connection_string,
+        )
+
+        return client
+    
+    # for debugging purposes: if a test fails and its agent has not been deleted, it will continue to show up in the agents list
+    """
+    # NOTE: this test should not be run against a shared resource, as it will delete all agents
+    @agentClientPreparer()
+    @recorded_by_proxy
+    def test_clear_client(self, **kwargs):
+        # create client
+        client = self.create_client(**kwargs)
+        assert isinstance(client, AIProjectClient)
+
+        # clear agent list
+        agents = client.agents.list_agents().data
+        for agent in agents:
+            client.agents.delete_agent(agent.id)
+        assert client.agents.list_agents().data.__len__() == 0
+       
+        # close client
+        client.close()
+    """
+
+    # # test agent creation and deletion
+    # @agentClientPreparer()
+    # @recorded_by_proxy
+    # def test_create_delete_agent(self, **kwargs):
+    #     # create client
+    #     client = self.create_client(**kwargs)
+    #     assert isinstance(client, AIProjectClient)
+    #     print("Created client")
+
+    #     # create agent
+    #     agent = client.agents.create_agent(model="gpt-4o", name="my-agent", instructions="You are helpful agent")
+    #     assert agent.id
+    #     print("Created agent, agent ID", agent.id)
+        
+    #     # delete agent and close client
+    #     client.agents.delete_agent(agent.id)
+    #     print("Deleted agent")
+    #     client.close()
+
+    # **********************************************************************************
+    #
+    #                               UNIT TESTS
+    #
+    # **********************************************************************************
+
+    # # **********************************************************************************
+    # #
+    # #                      HAPPY PATH SERVICE TESTS - File Search
+    # #
+    # # **********************************************************************************
+
+    # TODO test upload file
+
+    # test assistant creation with file_search and add_vector_store
+    @agentClientPreparer()
+    @recorded_by_proxy
+    def test_file_search_add_vector_store(self, **kwargs):
+        # Create client
+        client = self.create_client(**kwargs)
+        assert isinstance(client, AIProjectClient)
```

Comment:
Please also have some (or all) of the tests use the "with ... as ..." pattern, to test calls to `__enter__` and `__exit__` methods. So:
`with self.create_client() as client:`

In this case you don't need an explicit call to `client.close()` at the end